ors-api/src/main/java/org/heigit/ors/api/controllers/ExportAPI.java

```diff
@@ -127,6 +128,32 @@ public JsonExportResponse getJsonExport(
         return new JsonExportResponse(result);
     }
 
+    @PostMapping(value = "/{profile}/topojson", produces = {"application/json;charset=UTF-8"})
+    @Operation(
+            description = "Returns a list of edges, edge properties, and their topology within a given bounding box for a selected profile.",
+            summary = "Export Service TopoJSON"
+    )
+    @ApiResponse(
+            responseCode = "200",
+            description = "TopoJSON Response.",
+            content = {@Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = TopoJsonExportResponse.class)
+            )
+            })
+    public TopoJsonExportResponse getTopoJsonExport(
+            @Parameter(description = "Specifies the profile.", required = true, example = "driving-car") @PathVariable String profile,
+            @Parameter(description = "The request payload", required = true) @RequestBody ExportApiRequest request) throws StatusCodeException {
+        request.setProfile(getProfileEnum(profile));
+        request.setProfileName(profile);
+        request.setResponseType(APIEnums.ExportResponseType.TOPOJSON);
+
+        ExportResult result = exportService.generateExportFromRequest(request);
```

Comment:
in both cases that function gets data from the graph for a specific bbox, the function below does different things with that result, I don't think that's so confusing...

Reply:
I agree with both of you :)
The main issue is burried deeper in the software design that spreads the computation logic across several levels of abstraction and differently for each end-point. This will become better, once the refactorings of RoutingProfile and RoutingProfileManager are fully done.